src/Microsoft.AspNet.OData.Shared/Builder/DerivedTypeConstraintAttribute.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using Microsoft.AspNet.OData.Common;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.OData.Builder
+{
+    /// <summary>
+    /// Represents an <see cref="Attribute"/> that can be placed on a property or placed on a class to specify the derived type constraints. 
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property)]
+    public sealed class DerivedTypeConstraintAttribute : Attribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DerivedTypeConstraintAttribute"/> class.
+        /// </summary>
+        public DerivedTypeConstraintAttribute()
+        {
+            DerivedTypeConstraints = new HashSet<Type>();
+        }
```

Comment:
why do we have a empty here?

Reply:
Required by the compiler. Need to have at least one constructor with CLS compliant parameters for these classes. 

---
In reply to: [394517978](https://github.com/OData/WebApi/pull/1977#discussion_r394517978) [](ancestors = 394517978)